packages/yoastseo/src/languageProcessing/helpers/sentence/getSentences.js

```diff
@@ -39,16 +35,23 @@ const getSentencesFromBlockCached = memoize( getSentenceTokenizer );
  * @returns {Array} Sentences found in the text.
  */
 export default function( text ) {
+	// Remove Table of Contents.
 	text = excludeTableOfContentsTag( text );
-	text = unifyWhitespace( text );
+	// Unify whitespaces and non-breaking spaces.
+	text = unifyNonBreakingSpace( text );
+
 	let blocks = getBlocks( text );
 
 	// Split each block on newlines.
 	blocks = flatMap( blocks, function( block ) {
 		return block.split( newLineRegex );
 	} );
 
-	const sentences = flatMap( blocks, getSentencesFromBlockCached );
+	let sentences = flatMap( blocks, getSentencesFromBlockCached );
+	const paragraphTagsRegex = new RegExp( "^(<p>|</p>)$" );
+	// Filter sentences that contain only paragraph tags.
+	// This step is necessary since switching between editors might add extra paragraph tags that are not properly sanitized in the previous steps.
+	sentences = filter( sentences, sentence => ! paragraphTagsRegex.test( sentence ) );
```

Comment:
For the reviewers:

Here, the sentences that contains only paragraph tags are filtered out. This is necessary since the added paragraph tags when exiting Elementor editor after an edit, are split into individual sentences. If you know a better or cleaner way of doing this, feel free to leave a comment :) 

Another approach that I tried is by using the `stripTags` helper in each sentence. However, this leads to an undesired changes where a sentence like this: `"Obviously, 20.0 < 25.0 and 50.0 > 30.0."` is further sanitized further into `"Obviously, 20.0 30.0."`.